app/views/thredded/shared/_topic_nav.html.erb

```diff
@@ -2,19 +2,21 @@
   <%= yield :thredded_breadcrumbs %>
 
   <ul class="thredded--topic-navigation">
-    <% if messageboard.try(:persisted?) %>
+    <% if messageboard.try(:persisted?) -%>
       <li class="thredded--topic-navigation--public">
-        <a href="<%= messageboard_topics_path(messageboard) %>">Topics</a>
+        <%= link_to 'Topics', messageboard_topics_path(messageboard) %>
       </li>
-    <% end %>
+    <% end -%>
+
     <li class="thredded--topic-navigation--private">
-      <a href="<%= private_topics_path %>">Private Topics
-        <% if (unread_count = unread_private_topics_count) > 0 %>
-            <span class="thredded--topic-navigation--unread">
-              <%= unread_private_topics_count %>
-            </span>
-        <% end %>
-      </a>
+      <%= link_to private_topics_path do %>
+        Private Topics
+        <% if unread_private_topics_count > 0 -%>
+          <span class="thredded--topic-navigation--unread">
+            <%= unread_private_topics_count %>
```

Comment:
Ah I knew I forgot something. I meant to memoize the value in the
controller method to accomplish the reduction of extra method calls.

Also, using the `unread_count` in here is an extra level of redirection
that I feel is a bit unnecessary.

On Sunday, March 27, 2016, Gleb Mazovetskiy <notifications@github.com>
wrote:

> In app/views/thredded/shared/_topic_nav.html.erb
> <https://github.com/thredded/thredded/pull/166#discussion_r57530117>:
>
> >      <li class="thredded--topic-navigation--private">
> > -      <a href="<%= private_topics_path %>">Private Topics
> > -        <% if (unread_count = unread_private_topics_count) > 0 %>
> > -            <span class="thredded--topic-navigation--unread">
> > -              <%= unread_private_topics_count %>
> > -            </span>
> > -        <% end %>
> > -      </a>
> > +      <%= link_to private_topics_path do %>
> > +        Private Topics
> > +        <% if unread_private_topics_count > 0 -%>
> > +          <span class="thredded--topic-navigation--unread">
> > +            <%= unread_private_topics_count %>
>
> Would be better to use unread_count here to avoid an extra query here
> (it's cached in the query cache, but still).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/thredded/thredded/pull/166/files/1d76c31f52fd83f850436bb5231c1bcd44d71d6a#r57530117>
>